File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq
def solution(scoville, K):
    heapq.heapify(scoville)      
    answer = 0
    
    while len(scoville)>=2 and scoville[0]<K:
        if all(sco >=K for sco in scoville):
            return answer
        else:
            first = heapq.heappop(scoville)
            second = heapq.heappop(scoville)
            new_scoville = first + second * 2
            heapq.heappush(scoville,new_scoville)
            answer += 1
    if len(scoville)<=1 and scoville[0]<K:
        return - 1
    return answer

File: 프로그래머스/2/42626. 더 맵게/더 맵게.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;

int solution(vector<int> scoville, int K) {
    int answer = 0;
    int first, second, tmp;
    priority_queue<int,vector<int>,greater<int>> pq(scoville.begin(),scoville.end());
    while (pq.size() > 1 && pq.top() < K) {
        first = pq.top();
        pq.pop();
        second = pq.top();
        pq.pop();
        tmp = first + 2 * second;
        pq.push(tmp);
        answer++;
    }
    if (pq.top() < K) {
        return -1 ;
    }
    return answer;
}